src/uct/sm/mm/mm_iface.c

```diff
@@ -248,8 +253,13 @@ static inline void uct_mm_iface_poll_fifo(uct_mm_iface_t *iface)
 static void uct_mm_iface_progress(void *arg)
 {
     uct_mm_iface_t *iface = arg;
+    unsigned send_attempts = iface->config.send_attempts;
 
+    /* progress receive */
     uct_mm_iface_poll_fifo(iface);
+
+    /* progress the pending sends (if there are any) */
+    ucs_arbiter_dispatch(&iface->arbiter, 1, uct_mm_ep_process_pending, &send_attempts);
```

Comment:
I'm curious about the overhead of this check. Did you compare latency with and without the call ?